src/Sdk/WebApi/WebApi/Results/ResultsHttpClient.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.Services.Results.Contracts;
+using System.Net.Http.Formatting;
+using Sdk.WebApi.WebApi;
+
+namespace GitHub.Services.Results.Client
+{
+    public class ResultsHttpClient : RawHttpClientBase
+    {
+        public ResultsHttpClient(
+            Uri baseUrl,
+            HttpMessageHandler pipeline,
+            string token,
+            bool disposeHandler)
+            : base(baseUrl, pipeline, disposeHandler)
+        {
+            m_token = token;
+            m_resultsServiceUrl = baseUrl;
+            m_formatter = new JsonMediaTypeFormatter();
+        }
+
+        public async Task<GetSignedStepSummaryURLResponse> GetStepSummaryUploadUrlAsync(string jobId, string planId, string stepId, CancellationToken cancellationToken)
+        {
+            var request = new GetSignedStepSummaryURLRequest()
+            {
+                workflow_job_run_backend_id = jobId,
+                workflow_run_backend_id = planId,
+                step_backend_id = stepId
+            };
+
+            var stepSummaryUploadRequest = m_resultsServiceUrl+"twirp/results.services.receiver.Receiver/GetStepSummarySignedBlobURL";
+
+            HttpRequestMessage requestMessage = new HttpRequestMessage(HttpMethod.Post, stepSummaryUploadRequest);
+            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", m_token);
+            requestMessage.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json"));
+
+            HttpContent content = new ObjectContent<GetSignedStepSummaryURLRequest>(request, m_formatter);
+            requestMessage.Content = content;
+
+            var response = await SendAsync(requestMessage, HttpCompletionOption.ResponseContentRead, cancellationToken: cancellationToken);
+            return await ReadJsonContentAsync<GetSignedStepSummaryURLResponse>(response, cancellationToken);
+        }
+
+        public async Task StepSummaryUploadCompleteAsync(string jobId, string planId, string stepId, long size, CancellationToken cancellationToken)
+        {
+            var timestamp = DateTime.UtcNow.ToString("yyyy-MM-dd'T'HH:mm:ss.fffK");
```

Comment:
we talked about this in dm's, but our proto uses timestamp which is seconds and fractions of seconds at utc epoch

```json
"uploaded_at": {
   "seconds": 1670271570,
   "nanos": 110206903
}
```

it'd be great to use that for consistency instead of relying on string parsing + rfc3339